File: common/src/main/java/dev/qixils/aviary/common/messaging/AbstractPluginChannel.java
package dev.qixils.aviary.common.messaging;

/**
 * A utility class which implements some basic functionality of a plugin channel.
 */
public abstract class AbstractPluginChannel implements PluginChannel {
	private byte id = 0x00;

	@Override
	public byte nextId() {
		return id++;
	}
}

File: common/src/main/kotlin/dev/qixils/aviary/common/messaging/AbstractPluginChannel.kt
package dev.qixils.aviary.common.messaging

/**
 * A utility class which implements some basic functionality of a plugin channel.
 */
abstract class AbstractPluginChannel : PluginChannel {

	private var msgId: Byte = 0x00

	override fun nextId(): Byte {
		return msgId++
	}
}
